app/DynamicsAdapter/DynamicsAdapter.Web/Register/SearchRequestRegister.cs

```diff
@@ -50,6 +57,31 @@ public async Task<bool> RegisterSearchApiRequest(SSG_SearchApiRequest request)
             return true;
         }
 
+        public async Task<bool> RegisterDataProviders(SSG_DataProvider[] providers)
+        {
+            if (providers == null) return false;
+            await _cache.Save($"{Keys.REDIS_KEY_PREFIX}{Keys.DATA_PROVIDER_KEY}", providers, new TimeSpan(24, 0, 10));
+            return true;
+        }
+
+        public async Task<SSG_DataProvider[]> GetDataProvidersList()
+        {
+          
+            string data = await _cache.Get($"{Keys.REDIS_KEY_PREFIX}{Keys.DATA_PROVIDER_KEY}");
+            if (string.IsNullOrEmpty(data))
+            {
+                var cts = new CancellationToken();
+                var providers = await _oDataClient.For<SSG_DataProvider>()
```

Comment:
actually, all the real data exchanging with Dynamics should happen in FAMS3Adapter.Dynamics.
you can add a GetDataProviders in SearchRequestService or Add DataProviderService (in FAMS3Adapter.Dynamics.).
this register only has reference with Redis.

Reply:
Okay, will leave as is. I struggle with another depedency but i understand the need to have a single place for all the SSG call related calls. there are in a bunch of places and I would want to add to SRService